include/picongpu/plugins/openPMD/WriteSpecies.hpp

```diff
@@ -398,39 +421,85 @@ namespace picongpu
                     {
-                        strategy->prepare(currentStep, T_SpeciesFilter::getName(), std::move(runParameters));
-                    }
-                    log<picLog::INPUT_OUTPUT>("openPMD:  (begin) write particle records for %1%")
-                        % T_SpeciesFilter::getName();
-
-                    meta::ForEach<
-                        typename NewParticleDescription::ValueTypeSeq,
-                        openPMD::ParticleAttribute<boost::mpl::_1>>
-                        writeToOpenPMD;
-                    writeToOpenPMD(
-                        params,
-                        hostFrame,
-                        particleSpecies,
-                        basename,
-                        myNumParticles,
-                        globalNumParticles,
-                        myParticleOffset);
+                        ChunkDescription chunk;
+                        if(dumpIteration < particleIoChunkInfo.ranges.size())
+                        {
+                            chunk = particleIoChunkInfo.ranges[dumpIteration];
+                        }
+
+                        RunParameters_T runParameters(
+                            dc,
+                            *params,
+                            speciesTmp,
+                            filter,
+                            particleFilter,
+                            particleToTotalDomainOffset,
+                            globalNumParticles);
+
+                        if(chunk.numberOfParticles > 0)
+                        {
+                            strategy->prepare(
+                                currentStep,
+                                T_SpeciesFilter::getName(),
+                                runParameters,
+                                chunk.numberOfParticles,
+                                chunk.beginSupercellIdx,
+                                chunk.endSupercellIdx);
+                        }
+                        log<picLog::INPUT_OUTPUT>("openPMD: (begin) write particle records for %1%, dumping round %2%")
+                            % T_SpeciesFilter::getName() % dumpIteration;
+
+                        meta::ForEach<
+                            typename NewParticleDescription::ValueTypeSeq,
+                            openPMD::ParticleAttribute<boost::mpl::_1>>
+                            writeToOpenPMD;
+                        writeToOpenPMD(
+                            params,
+                            hostFrame,
+                            particleSpecies,
+                            basename,
+                            chunk.numberOfParticles,
+                            globalNumParticles,
+                            myParticleOffset + particleOffset);
+
+                        particleOffset += chunk.numberOfParticles;
+                        ++dumpIteration;
+                        log<picLog::INPUT_OUTPUT>("openPMD: (end) write particle records for %1%, dumping round %2%")
```

Comment:
Comment must be moved two lines up to have the correct iteration in the log message.